### DISCUSSION TITLE
Package installation error

### DISCUSSION THREAD

**Discussion by viniciusweber** [imdhemy/laravel-in-app-purchases#31] [Q&A]
I was unable to resolve the error when installing the package on Laravel 7.2.0

Any tips on how to solve?

```

> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   ErrorException

  array_merge_recursive(): Expected parameter 2 to be an array, null given

  at \vendor\laravel\framework\src\Illuminate\Foundation\Support\Providers\EventServiceProvider.php:69
    65|             return $cache[get_class($this)] ?? [];
    66|         } else {
    67|             return array_merge_recursive(
    68|                 $this->discoveredEvents(),
  > 69|                 $this->listens()
    70|             );
    71|         }
    72|     }
    73|

  1   \vendor\laravel\framework\src\Illuminate\Foundation\Support\Providers\EventServiceProvider.php:69
      array_merge_recursive([])

  2   \vendor\laravel\framework\src\Illuminate\Foundation\Support\Providers\EventServiceProvider.php:32
      Illuminate\Foundation\Support\Providers\EventServiceProvider::getEvents()
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Comment by imdhemy** [comment]
Hi @viniciusweber,

I just installed the package on a fresh Laravel installation `^7.29`, and it works as expected. 

Did you tried to
1. Clear the config cache? 
`php artisan config:clear`
2. Did you tried to install the package on a freshly installed laravel?
```
composer create-project laravel/laravel laravel_7_app "7.*.*"
composer require imdhemy/laravel-purchases
```

**Reply by viniciusweber** [reply]
Sorry for the delay, I tried to install it in an ongoing project.

**Comment by denisakira** [comment]
I'm having this problem too. 

Installing the package in a fresh Laravel project works as expected, but in my project it happens.

----

EDIT: I was able to solve it by manually deleting the `bootstrap/cache/packages.php` and `bootstrap/cache/services.php` files.

**Reply by imdhemy** [reply]
Awesome, thanks @denisakira

**Reply by viniciusweber** [reply]
Thanks for the feedback, I want to use it in another project in progress and I will test your tip, in the current one I ended up using a simpler package.
